targets/XPlatCppSdk/source/code/source/playfab/PlayFabIOSHttpPlugin.mm

```diff
@@ -0,0 +1,346 @@
+        httpRequestMutex.unlock();
+    }
+
+    void PlayFabIOSHttpPlugin::MakePostRequest(std::unique_ptr<CallRequestContainerBase> requestContainer)
+    {
+        std::shared_ptr<RequestTask> requestTask = nullptr;
+        try
+        {
+            requestTask = std::make_shared<RequestTask>();
+            requestTask->Initialize(requestContainer);
+        }
+        catch (const std::exception& ex)
+        {
+            PlayFabPluginManager::GetInstance().HandleException(ex);
+        }
+        catch (...)
+        {
+
+        }
+        if(requestTask != nullptr)
+        { // LOCK httpRequestMutex
+            std::unique_lock<std::mutex> lock(httpRequestMutex);
+            requestTask->state = RequestTask::Pending;
+            pendingRequests.push_back(std::move(requestTask));
+            if(workerThread == nullptr)
+            {
+                threadRunning = true;
+                workerThread = std::make_unique<std::thread>(&PlayFabIOSHttpPlugin::WorkerThread, this);
+            }
+        } // UNLOCK httpRequestMutex
+    }
+
+    size_t PlayFabIOSHttpPlugin::Update()
+    {
+        if (PlayFabSettings::threadedCallbacks)
+        {
+            throw std::runtime_error("You should not call Update() when PlayFabSettings::threadedCallbacks == true");
+        }
+
+        std::shared_ptr<RequestTask> requestTask = nullptr;
+        { // LOCK httpRequestMutex
+            std::unique_lock<std::mutex> lock(httpRequestMutex);
+            if (pendingResults.empty())
+            {
+                return pendingRequests.size();
+            }
+
+            requestTask = std::move(this->pendingResults[0]);
+            this->pendingResults.pop_front();
+        } // UNLOCK httpRequestMutex
+
+        HandleResults(*requestTask);
+
+        // activeRequestCount can be altered by HandleResults, so we have to re-lock and return an updated value
+        { // LOCK httpRequestMutex
+            std::unique_lock<std::mutex> lock(httpRequestMutex);
+            return pendingRequests.size() + pendingRequests.size();
+        }
+    }
+
+    void PlayFabIOSHttpPlugin::WorkerThread()
+    {
+        { // LOCK httpRequestMutex
+            std::unique_lock<std::mutex> lock(httpRequestMutex);
+            this->requestingTask = nullptr;
+        }
+        while(threadRunning)
+        {
+            RequestTask::State state;
+            { // LOCK httpRequestMutex
+                std::unique_lock<std::mutex> lock(httpRequestMutex);
+                if(this->requestingTask == nullptr)
+                {
+                    if (!pendingRequests.empty())
+                    {
+                        this->requestingTask = this->pendingRequests[0];
+                        this->pendingRequests.pop_front();
+                    }
+                    else
+                    {
+                        threadRunning = false;
+                        workerThread->detach();
+                        workerThread = nullptr;
```

Comment:
If there are no pending requests then this will exit the thread, is that correct? and then when a new PostRequest comes we will start a new thread? if yes then is it not expensive to start a new thread for every request?